### PR TITLE
Update contact form to use API Gateway endpoint to invoke Lambda func…

### PR DESCRIPTION
**Pull request by emilyemilyemilyemilyemilyemily**
…tion for submissions

### Ticket(s)

https://airtable.com/appfJZShN8K4tcWHU/pagXdnDYi0tVl4u8R?HjEAY=recO4YtbcejXzqKGR

### Type of change

- [x ] Design change

### Description

Update contact form to use API Gateway endpoint to invoke Lambda function for submissions. 

On submit, POST -> [API Gateway](https://us-west-2.console.aws.amazon.com/apigateway/main/apis/gv6n06a4h2/resources?api=gv6n06a4h2&experience=rest&region=us-west-2) -> [contact-form Lambda](https://us-west-2.console.aws.amazon.com/lambda/home?region=us-west-2#/functions/contact-form?newFunction=true&tab=code) -> send email to info@cv via AWS SES.

Also add success message upon successful submission.

#### Preview
Use your local build.

- [x ] Everything works as expected
- [x ] Check changed page(s)
- [x ] Check large screen size (>1400)
- [x ] Check tablet screen size (<786)
- [x ] Check mobile screen size (<375)
- [ x] Check that links work
- [ x] Check the console for errors or warnings
- [ x] Check the terminal for any errors or problems
- [ x] Check for any uncaught ESLint problems

#### Code Clarity and Readability
- [ x] Are variable and function names descriptive, meaningful, in the appropriate case (ex: snake, camel)?
- [x ] Is the code self-explanatory and easy to understand?
- [x ] Are comments used where necessary to explain complex logic or provide additional context?

#### Code Efficiency
- [ x] Is the code DRY?
- [ x] Are there any areas of unnecessary complexity or duplication that might benefit from refactoring?

#### Code Modularity and Reusability
- [x ] Is the code and any new components organized in a logical location in the directory?
- [x ] Can any code snippets be refactored into reusable components?
- [x ] Are there opportunities to abstract common functionality into separate functions or classes?
- [x ] Is there good use of CSS selectors when creating new components or refactoring old ones?

#### Best Practices and Coding Standards
- [x ] Does the code follow the established coding standards and style guide of the project?
- [ x] Are there any violations of best practices that should be addressed?
- [ x] Are there any hard-coded strings or UI elements that should be externalized?
- [ x] Are all measurement units in REM? (border-radius, borders should still be in px)
- [ x] Is there any unnecessary commented out code lingering?
- [ x] Are all files TypeScript?

### Overall PR Checklist:

_Please delete any options that are not relevant_

- [x ] I have performed a self-review of my own code
- [x ] My changes generate no new warnings in the console
- [x ] There are no new linting errors/problems in my code
- [x ] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

**Review by sramalho94 — CHANGES_REQUESTED** [review]
hey @emilyemilyemilyemilyemilyemily , great work with this TG so excited to be using lambda functions for this. 

Could you switch the target branch to `staging` pls, we can merge the new functionality into `main` after the team has reconvened on Tuesday.

Do you think we should include the new `lambda` directory code in this PR, or is it better practice to just keep it on a separate branch?

File: pages/contact.tsx [hunk 1]
```diff
@@ -71,9 +75,31 @@ export default function ContactPage() {
...
+        const errorText = await response.text();
+        setFormError(`Server error: ${errorText}`);
+      }
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        setFormError(`Network error: ${error.message}`);
+      } else {
+        setFormError('An unknown error occurred');
+      }
     }
```
Comment (LeoFuna) [inline pages/contact.tsx:103]: Hey @emilyemilyemilyemilyemilyemily
Can I suggest one thing here? 
This logic that handle our request doesnt seems to be a `contact.tsx` page`s responsible.

We are adding logic here that increase our complexity.
What do you think about using a `custom hook` to help us to separate that request logic and just use `methods` and `properties` that are exposed by that hook?

I created an example to make this more clear [here](https://gist.github.com/LeoFuna/f85be3cb21ac57a08cd007017b5ee052)
```suggestion
    makeRequest(toSend);
```

File: pages/contact.tsx [hunk 2]
```diff
@@ -26,6 +28,7 @@ export default function ContactPage() {
   const [toSend, setToSend] = useState<FormData>(initialState);
   const [formError, setFormError] = useState<string>('');
   const [formErrors, setFormErrors] = useState<FormData>(initialState);
+  const [formSuccess, setFormSuccess] = useState<string>('');
```
Comment (LeoFuna) [inline pages/contact.tsx:31]:
```suggestion
  const { makeRequest, isSuccess, errorMessage } = useSubmitAWSForm();

  useEffect(() => {
    if (isSuccess) setFormError('');
    if (errorMessage) setFormError(errorMessage);

  }, [isSuccess, errorMessage]);
```
If we create a custom hook, I believe that we dont need this anymore.

File: pages/contact.tsx [hunk 3]
```diff
@@ -215,6 +239,12 @@ export default function ContactPage() {
             {formError}
           </FormHelperText>
           )}
+
+          {formSuccess && (
+          <FormHelperText id="formSuccess" aria-live="polite" style={{ color: 'green', fontSize: '1.1rem' }}>
+            {formSuccess}
```
Comment (LeoFuna) [inline pages/contact.tsx:245]:
```suggestion
          {isSuccess && (
          <FormHelperText id="formSuccess" aria-live="polite" style={{ color: 'green', fontSize: '1.1rem' }}>
            Your message was sent successfully!
```

**Review by LeoFuna — CHANGES_REQUESTED** [review]
Thx for helping in this Lambda thing! :partying_face: 

Thinking about separation of responsability, i would suggest to remove from this page that request logic, that helps a lot on maintenance.
Let me know what do you think

File: pages/contact.tsx [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 
+import usePostRequest from '../components/functional/usePostRequest.tsx';
 import IndividualPageHead from '../components/helper/IndividualPageHead.tsx';
 import ImageContainer from '../components/layout/ImageContainer.tsx';
 import SectionContainer from '../components/layout/SectionContainer.tsx';
 import content from '../content/contact.ts';
 
+const API_ENDPOINT = 'https://gv6n06a4h2.execute-api.us-west-2.amazonaws.com/Prod/contact-form';
```
Comment (e-papanicolas) [inline pages/contact.tsx:12]: @emilyemilyemilyemilyemilyemily is the API endpoint something we should store in `.env` or is it okay to be public?

**Comment by e-papanicolas** [comment]
@emilyemilyemilyemilyemilyemily great work, this is looking so good! I think that you have resolved everyone's comments, I just had a question about the API endpoint, but other than that this is ready to merge. I tested and confirmed the email was received! Very exciting. Can you also open a PR for the lambda function? 🙏🏼

File: pages/contact.tsx (hunk 4 above)
Comment (emilyemilyemilyemilyemilyemily) [reply pages/contact.tsx:12]: i think its fine to be here - it's a publicly accessible endpoint anyway and having it live in the code makes it easier for devs to test changes

Comment (e-papanicolas) [reply pages/contact.tsx:12]: Great!

**Comment by e-papanicolas** [comment]
@clearviction-devs/reviewers i am going to dismiss everyone's reviews to merge this. All of the comments were resolved.

**Pull request merged**